lib/EarlyFunctionAnalysis/DetectABI.cpp

```diff
@@ -99,6 +101,18 @@ static pipeline::RegisterAnalysis<DetectABIAnalysis> A1;
 
 namespace efa {
 
+static bool isWritingToMemory(llvm::Instruction &I) {
+  if (auto *StoreInst = dyn_cast<llvm::StoreInst>(&I)) {
+    auto *Destination = StoreInst->getPointerOperand();
+
+    if (auto *GV = dyn_cast<llvm::GlobalVariable>(Destination)) {
```

Comment:
You need to just use `isMemory`, no need to test `GlobalVariable`, it doesn't make much sense, see the implementation of `isMemory`.